spring-boot/admin/src/main/java/eu/coatrack/admin/service/GatewayConfigFilesStorage.java

```diff
@@ -0,0 +1,104 @@
+import java.nio.file.Paths;
+
+import eu.coatrack.api.Proxy;
+import eu.coatrack.api.ServiceApi;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+
+/**
+ *
+ * @author perezdf, ChristophBaier
+ */
+
+@Slf4j
+@Component
+public class GatewayConfigFilesStorage {
+
+    @Value("${proxy.config.files.folder}")
+    private String gatewayConfigFilesFolderLocation;
+
+    @Value("${ygg.admin.api-base-url-for-gateway}")
+    private String adminApiBaseUrlForGateway;
+
+    @Value("${is-run-via-docker-compose-setup}")
+    private boolean isDockerComposeSetup;
+
+    @PostConstruct
+    private void emptyProxyConfigFilesFolderIfExistsForNonDockerComposeDeployments() throws IOException {
+        if (!isDockerComposeSetup){
+            Path targetFolderPath = Paths.get(System.getProperty("user.dir") + "/target");
+            if (Files.notExists(targetFolderPath))
+                Files.createDirectory(targetFolderPath);
+
+            Path gatewayConfigFilesFolderPath = Paths.get(gatewayConfigFilesFolderLocation);
+            if (Files.exists(gatewayConfigFilesFolderPath)){
+                FileUtils.deleteDirectory(gatewayConfigFilesFolderPath.toFile());
+            }
+            Files.createDirectory(gatewayConfigFilesFolderPath);
+        }
+    }
+
+    public void addGatewayConfigFile(Proxy proxy) throws IOException {
+        PrintWriter writer = new PrintWriter(gatewayConfigFilesFolderLocation + "/ygg-proxy-" + proxy.getId() + ".yml", "UTF-8");
+        writer.println("proxy-id: " + proxy.getId());
+        writer.println("ygg.admin.api-base-url: " + adminApiBaseUrlForGateway);
+        if (proxy.getPort() != null) {
+            writer.println("server.port: " + proxy.getPort());
+        }
+        for (ServiceApi service : proxy.getServiceApis()) {
+            writer.println("zuul.routes." + service.getUriIdentifier() + ".url : " + service.getLocalUrl());
+        }
+        writer.println("zuul.host.connect-timeout-millis: 150000");
+        writer.println("zuul.host.socket-timeout-millis: 150000");
+        writer.close();
+    }
+
+    public void deleteGatewayConfigFile(Proxy proxy) throws IOException {
+        Path gatewayConfigFileToBeDeletedPath = Paths.get(gatewayConfigFilesFolderLocation + "/ygg-proxy-" + proxy.getId() + ".yml");
+        if (Files.exists(gatewayConfigFileToBeDeletedPath)) {
+            tryToDeleteGatewayConfigFile(proxy, gatewayConfigFileToBeDeletedPath);
+        } else {
+            throw new FileNotFoundException("Tried to delete the configuration file for proxy " + proxy.getId()
+                    + ", but there is no according file + " + gatewayConfigFileToBeDeletedPath);
+        }
+    }
+
+    private void tryToDeleteGatewayConfigFile(Proxy proxy, Path gatewayConfigFileToBeDeletedPath) throws IOException {
```

Comment:
This method does not really need the proxy object, just uses it for logging. Thus maybe remove this parameter?